services/src/main/java/com/linkedin/kmf/common/TopicFactory.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.common;
+
+/**
+ * Constructs the monitored topic if it does not exist.
+ *
+ * Implementations of this class should have a public constructor with the following signature: <br/>
+ *   Constructor(Map<String, ?> config) where config are additional configuration parameters passed in from the Kafka
+ *   Monitor configuration.
+ */
+public interface TopicFactory {
+
+  /**
+   * Creates the specified topic if it does not exist.
+   * @param zkUrl zookeeper connection url
+   * @param topic topic name
+   * @param replicationFactor the replication factor for the topic
+   * @param partitionToBrokerRatio This is multiplied by the number brokers to compute the number of partitions in the topic.
+   * @return The number of partitions for the specified topic.
+   */
+
+  public int createTopicIfNotExist(String zkUrl, String topic, int replicationFactor,
```

Comment:
It is redundant to specify `public` here. Can you remove it?

I actually think it is useful to have a properties parameter for this method? Currently min.isr = 1. When we move to JBOD we need to use min.isr = 2 when creating topic. We probably want to do it now to avoid changing interface in the future. It is not needed now so we can assign null as its value when this method is called.